lib/rails/generators/locales/component_generator.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require "rails/generators/abstract_generator"
+
+module Locales
+  module Generators
+    class ComponentGenerator < ::Rails::Generators::NamedBase
+      include ViewComponent::AbstractGenerator
+
+      source_root File.expand_path("templates", __dir__)
+      class_option :sidecar, type: :boolean, default: false
+
+      def create_locale_file
+        I18n.available_locales.each do |locale|
+          dest_with_path = destination(locale)
+          template "component.yml", dest_with_path
+          gsub_file dest_with_path, /\Aen:\n/, "#{locale}:\n"
```

Comment:
nit: gsub can become fragile over time, thoughts on using an instance variable instead?

```rb
I18n.available_locales.each do |locale|
  @locale = locale
  template "component.yml", destination(locale)
end
```

And the locale template be `component.yml.tt`:

```erb
<%= @locale %>:
  hello: "Hello world!"
```

Alternatively it could also dump a hash to YAML:

```rb
create_file "component.#{locale}.yml", { locale.to_s => { "hello" => "Hello world!" } }.to_yaml
```